qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -173,12 +172,18 @@ def retrieve_jobs(self, provider: AccountProvider, refresh: bool = False) -> Non
         jobs_dict = {}
         for job in jobs:
             # Verify the job is proper.
-            matched = pattern.match(job.name()) if job.name() else None
-            if not matched or matched.group(1) != self._name or \
-                    job.backend().name() != self._backend.name():
+            matched = JOB_SET_NAME_RE.match(job.name()) if job.name() else None
+            if not matched or job.backend().name() != self._backend.name():
```

Comment:
The idea was that the job name should at least follow the job set name format. But, I could see that it's not really necessary, since the job name could be changed at any time.